### PR TITLE
strip SERVICE_AUTH_TOKEN= from token file

### PR DESCRIPTION
**Pull request by spacejam**

### CODE REVIEW

File: src/mesos_state_client.erl [hunk 1]
```diff
@@ -61,20 +61,34 @@ try_token_from_file() ->
...
--spec(strip_token_binary(binary()) -> string()).
-strip_token_binary(Binary) ->
-  Stripped = binary:replace(Binary, ?WHITESPACE, <<>>, [global]),
-  binary_to_list(Stripped).
+-spec(parse_token(binary()) -> {ok, string()} | {error, no_token_parsed}).
+parse_token(<<>>) ->
+  {ok, Path} = application:get_env(?APP, ssl_token_file),
+  lager:error("failed to parse auth token from configured file: ~p", [Path]),
+  {error, no_token_parsed};
+parse_token(B) ->
```
Comment (sargun) [inline src/mesos_state_client.erl]: The idea was to make this more strict, not less strict.

If we're trying to be compatible with bash:

`os:cmd(". tyler && echo $FOO") -> "bar\n". -- where tyler:
```
FOO=bar
CAT=dog
```

Comment (sargun) [inline src/mesos_state_client.erl]:
```
parse_token(B0) -> 
B1 = binary:replace(B0, ?WHITESPACE, <<>>, [global]),
[<<"SERVICE_AUTH_TOKEN">>, Token] = binary:split(B1, <<"=">>),
Token.
```

File: src/mesos_state_client.erl [hunk 2]
```diff
@@ -61,20 +67,34 @@ try_token_from_file() ->
...
--spec(strip_token_binary(binary()) -> string()).
-strip_token_binary(Binary) ->
-  Stripped = binary:replace(Binary, ?WHITESPACE, <<>>, [global]),
-  binary_to_list(Stripped).
+-spec(parse_token(binary()) -> {ok, string()} | {error, no_token_parsed}).
+parse_token(<<>>) ->
+  Path = application:get_env(?APP, ssl_token_file, test_path),
+  lager:error("failed to parse auth token from configured file: ~p", [Path]),
+  {error, no_token_parsed};
+parse_token(B) ->
```
Comment (sargun) [inline src/mesos_state_client.erl]: This code would bit simpler as: 
```
(B0) ->
B1 = binary:replace(B0, ?WHITESPACE, <<>>, [global]),
[<<"SERVICE_AUTH_TOKEN">>, Token] = binary:split(B1, <<"=">>)
```
If there was bad data in the file, people would see it in their crash log, and we'd be able to fix it.

Comment (spacejam) [inline src/mesos_state_client.erl]: this is written for parsing multi-line env var files for if that becomes reality

**Pull request merged**